apps/music/views/albums/view.js

```diff
@@ -76,14 +76,16 @@ AlbumsView.prototype.getAlbums = function() {
       return this.fetch('/api/albums/list')
         .then(response => response.json())
         .then((albums) => {
-          albums.forEach((album) => {
-            if (!album.metadata.album) {
-              album.metadata.album = unknownAlbum;
-              album.unknownAlbum = true;
-            }
+          return albums.map((album) => {
+            return {
+              name: album.name,
+              url: '/album-detail?id=' + encodeURIComponent(album.name),
+              album: album.metadata.album || unknownAlbum,
+              unknownAlbum: !album.metadata.album ? true : false,
+              artist: album.metadata.artist
+            };
```

Comment:
I think a comment explaining what/why this function is doing what it's doing would be nice :)